code/frontend/src/components/Answer/AnswerParser.tsx

```diff
@@ -7,23 +7,32 @@ type ParsedAnswer = {
     markdownFormatText: string;
 };
 
+let filteredCitations = [] as Citation[];
+
+// Define a function to check if a citation with the same Chunk_Id already exists in filteredCitations
+const isDuplicate = (citation: Citation) => {
+    return filteredCitations.some((c) => c.chunk_id === citation.chunk_id);
+};
+
 export function parseAnswer(answer: AskResponse): ParsedAnswer {
     let answerText = answer.answer;
     const citationLinks = answerText.match(/\[(doc\d\d?\d?)]/g);
 
     const lengthDocN = "[doc".length;
 
-    let filteredCitations = [] as Citation[];
+    filteredCitations = [] as Citation[];
     let citationReindex = 0;
     citationLinks?.forEach(link => {
         // Replacing the links/citations with number
         let citationIndex = link.slice(lengthDocN, link.length - 1);
         let citation = cloneDeep(answer.citations[Number(citationIndex) - 1]) as Citation;
-        if (!filteredCitations.find((c) => c.id === citationIndex)) {
+        if (!isDuplicate(citation) && !filteredCitations.find((c) => c.id === citationIndex)) {
           answerText = answerText.replaceAll(link, ` ^${++citationReindex}^ `);
           citation.id = citationIndex; // original doc index to de-dupe
           citation.reindex_id = citationReindex.toString(); // reindex from 1 for display
           filteredCitations.push(citation);
+        }else if (isDuplicate(citation) && !filteredCitations.find((c) => c.id === citationIndex)){
+            answerText= answerText.replaceAll(link, '')
```

Comment:
As discussed on the call, this needs to be replaced with the `citationReindex` value of the duplicated citation to stop it being removed completly.

Reply:
Resolved
![Uploading image.png…]()
![Uploading image.png…]()

Reply:
Resolved
<img width="700" alt="image" src="https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/169637670/9e907a87-7761-4e60-b60f-770c6483f153">

<img width="536" alt="image" src="https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/169637670/03930ad4-7a52-4bb5-8e99-6f9b5237f8cf">